builder_data.yaml

```diff
@@ -48,6 +48,72 @@ apt_archive_path: "/opt/release-archives-staging/apt"
 freight_archive_path: "/opt/tools/freight-archives/apt"
 yum_archive_path: "/opt/release-archives-staging/yum"
 downloads_archive_path: "/opt/release-archives-staging/downloads"
+
+# Starting with puppet7, apt repos were organized by puppet version: puppet7,
+# puppet6.
+# Examples:
+#  /opt/tools/freight/apt/puppet7
+#  /opt/tools/freight-nightlies/apt/puppet7-nightly
+# These handle that condition
+community_apt_repo_generate: |
+  (
+    flock --wait 600 270
+    keychain -k mine;
+    eval $(/usr/bin/keychain -q --agents gpg --eval __GPG_KEY__);
+    export GPG_TTY=$(tty);
+    if printf -- %s "__APT_PUPPET_REPOSITORIES__" | grep --quiet -- APT_PUPPET_REPOSITORIES; then
+      sudo --preserve-env freight cache --conf=/etc/freight.conf.d/community.conf &
```

Comment:
I'm not sure if we'll be able to use the existing config file for the new repo layout, or if we'll need to have a different freight config for the puppet7 config. I feel like I remember something about those configs being tied specifically to a single repo but I can't recall for sure

Reply:
I'm not seeing any repo-specific stuff in them. It might be safer to create new ones though. Here's an example of the contents of community.conf

```bash
# Example Freight configuration.

# Directories for the Freight library and Freight cache.  Your web
# server's document root should be `$VARCACHE`.
VARLIB="/opt/tools/freight"
VARCACHE="/opt/repository/apt"

# Default `Origin` and `Label` fields for `Release` files.
ORIGIN="Puppetlabs"
LABEL="Puppetlabs"

# Defaults to just i386 and amd64, this means we can deb up scripts.
ARCHS="i386 amd64 arm64 armel armhf powerpc sparc mips mipsel ppc64el all"

# Cache the control files after each run (on), or regenerate them every time (of
f)
CACHE="on"

# GPG key to use to sign repositories.  This is required by the `apt`
# repository provider.  Use `gpg --gen-key` (see `gpg`(1) for more
# details) to generate a key and put its email address here.
GPG="release@puppet.com"

# on to follow symbolic links in VARLIB to produce extra components
# in the cache directory or off to offer no special treatment.
SYMLINKS="on"
```

Reply:
ah yeah I think the `VARLIB` and `VARCACHE` are going to need updating, probably just appending a `puppet7` to the end of each of those.